### PR TITLE
Initial Agon port

### PR DESCRIPTION
**Pull request by sijnstra**
This is a port of ZealOS to the Agon Light platform. It loads from MOS using a special loader. This is the initial port, with improvements to come over time.

### CODE REVIEW

File: Kconfig [hunk 1]
```diff
@@ -11,6 +11,8 @@ mainmenu "Zeal 8-bit OS Configuration"
         string
         default "zeal8bit" if TARGET_ZEAL_8_BIT
         default "trs80" if TARGET_TRS_80
+        default "agon" if TARGET_AGON_LIGHT
+        default "example" if TARGET_EXAMPLE
```
Comment (Zeal8bit) [inline Kconfig:15]: The example target has been removed from the latest HEAD, you can remove this

File: README.md [hunk 2]
```diff
@@ -571,6 +572,21 @@ To have a more interesting port, the following features would need to be impleme
 * A disk to store the `init.bin`/romdisk, can be read-only, so can be stored on the ROM
 * A read-write disk to store data, can be a floppy disk driver using ZealFS filesystem
 
+## Agon Light
+
+A port to the eZ80 powered Agon Light, written and maintained by [Shawn Sijnstra](https://github.com/sijnstra/Zeal-8-bit-OS). Feel free to use that fork for Agon specific bugs/requests. This uses the non-MMU kernel, and implements as many of the features that the Zeal implementation supports.
```
Comment (Zeal8bit) [inline README.md:577]: Did you mean  `the Zeal 8-bit Computer implementation supports` or `the Zeal 8-bit OS implementation supports`?

File: target/agon/include/video_h.asm [hunk 3]
```diff
@@ -0,0 +1,80 @@
+; SPDX-FileCopyrightText: 2023 Zeal 8-bit Computer <contact@zeal8bit.com>
```
Comment (Zeal8bit) [inline target/agon/include/video_h.asm:1]: Do you use this file? You can get rid of it to simplify the port if you don't use it

**Review by Zeal8bit — COMMENTED** [review]
Thanks for your contribution!

Looks very good, I added a few remarks and I saw there are some binary files also. Can you please get rid of all the build directory `agon/romdisk/build` directory?

File: target/agon/include/video_h.asm (hunk 3 above)
Comment (sijnstra) [reply target/agon/include/video_h.asm:1]: As it happens I do use a few of the defines in here for video control modes.

File: README.md (hunk 2 above)
Comment (sijnstra) [reply README.md]: Updated and clarified (I hope!)

File: Kconfig (hunk 1 above)
Comment (sijnstra) [reply Kconfig]: Fixed

**Comment by sijnstra** [comment]
Those tweaks should do it. Let me know if you spot anything else that isn't quite right.

**Comment by Zeal8bit** [comment]
Thanks for fixing the few issues there was, merging!

**Pull request merged**